src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -839,7 +840,47 @@ public Task InitializeExtensionsAsync(IEnumerable<string> pathToAdditionalExtens
         ITestRunAttachmentsProcessingEventsHandler eventsHandler,
         CancellationToken cancellationToken)
     {
-        throw new NotImplementedException();
+        _testPlatformEventSource.TranslationLayerTestRunAttachmentsProcessingStart();
+
+        try
+        {
+            var attachmentProcessingPayload = new TestRunAttachmentsProcessingPayload
+            {
+                Attachments = attachments,
+                InvokedDataCollectors = invokedDataCollectors,
+                RunSettings = processingSettings,
+                CollectMetrics = collectMetrics
+            };
+
+            using (cancellationToken.Register(() =>
+                TestRequestManager?.CancelTestRunAttachmentsProcessing()))
+            {
+                // Awaiting the attachment processing task here. The implementation of the
+                // underlying operation guarantees the event handler is called when processing
+                // is complete, so when awaiting ends, the results have already been passed to
+                // the caller via the event handler. No need for further synchronization.
+                await Task.Run(() =>
+                        TestRequestManager?.ProcessTestRunAttachments(
+                            attachmentProcessingPayload,
+                            new InProcessTestRunAttachmentsProcessingEventsHandler(eventsHandler),
+                            new ProtocolConfig { Version = _highestSupportedVersion }),
+                        CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+        }
+        catch (Exception ex)
```

Comment:
I would really like for us to avoid always catching all when possible. I am not sure what is your goal here but if you just want to prevent the errors linked to cancellation, we could catch a narrow set of exceptions.

Reply:
Catching all possible exceptions here is by design and it's more of a safety net for defensive programming. All methods in the in-process wrapper follow this convention. With the out-of-process implementation we had before, no TP exceptions would have been able to cross the process boundary to TW via exception propagation. Instead, errors would be propagated via special methods on each of the event handlers. With the in-process implementation this is no longer true and TP exceptions could be propagated to TW code which may not be ready to handle them and nasty crashes could occur. Instead, I prefer to propagate the errors via the same old mechanism used by the out-of-process wrapper (i.e. special purpose methods on event handlers). This ensures both the safety and interchangeability of the two wrappers (in-proc vs out-of-proc), as well as preserving the old tried-and-tested error propagation mechanism.